src/context/context.tsx

```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+import { createContext, useEffect, useState } from "react";
+
+interface NewsContextInterface {
+    children: React.ReactNode;
+}
+
+export const NewsContext = createContext(undefined);
+
+export const NewsContextProvider = (props: NewsContextInterface) => {
+    const [data, setData] = useState();
+    const apikey = "e75c1dd786c8454d989a14ccd2e3ca27";
+    useEffect(() => {
+        axios
+            .get(
+                `https://newsapi.org/v2/top-headlines?country=us&pageSize=100&apiKey=${apikey}`
+            )
+            .then((r) => {
+                setData(r.data);
+            });
+    }, []);
```

Comment:
I like what you did here. But page size should be actual page size (10 or 20). And when the user clicks you should load more articles with the same page size count.

Reply:
You don't really have infinite scroll implementation here. You are just mocking this with 100 calls.

Reply:
I was hoping you are not going to notice it. But it is really sneaky, right?